modules/tensor_mechanics/include/materials/RankTwoCylindricalComponent.h

```diff
@@ -0,0 +1,43 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "Material.h"
+#include "RankTwoTensor.h"
+
+/**
+ * MaterialRankTwoCylindricalScalar uses the namespace RankTwoScalarTools to compute scalar
+ * values from Rank-2 tensors.
+ */
+class RankTwoCylindricalComponent : public Material
+{
+public:
+  static InputParameters validParams();
+
+  RankTwoCylindricalComponent(const InputParameters & parameters);
+
+protected:
+  virtual void initQpStatefulProperties() override;
+  virtual void computeQpProperties() override;
+
+  const MaterialProperty<RankTwoTensor> & _tensor;
```

Comment:
Add doxygen descriptions of variables. Use the one-liner form (`///`) for all of them for consistency.